src/ui/public/vis/editors/default/default.js

```diff
@@ -110,10 +115,15 @@ const defaultEditor = function ($rootScope, $compile) {
             }
           };
 
-          $scope.$watch(function () {
-            return $scope.vis.getCurrentState(false);
+          let lockDirty = false;
+          $scope.$watch(() => {
+            return $scope.vis.getSerializableState($scope.state);
           }, function (newState) {
-            $scope.vis.dirty = !angular.equals(newState, $scope.vis.getEnabledState());
+            if (lockDirty) {
+              lockDirty = false;
+            } else {
+              $scope.vis.dirty = !angular.equals(newState, $scope.oldState);
```

Comment:
It looks like we're using `lockDirty` to prevent `scope.vis.dirty` from being updated if there have been any changes to the `vis.state`. It's not clear to me why we're doing this though... could you add a comment to explain?

Reply:
when visualization changes its parameter (on `vis.params`) editor needs to be notified about the change and update the appropriate option, but its dirty state should not change as the setting was already applied on the `vis.params`

i will add a comment